smac/optimizer/ei_optimization.py

```diff
@@ -247,75 +252,139 @@ def _get_initial_points(self, num_points, runhistory):
-                self.logger.debug("Local search took %d steps and looked at %d "
-                                  "configurations. Computing the acquisition "
-                                  "value for one configuration took %f seconds"
-                                  " on average.",
-                                  local_search_steps, neighbors_looked_at,
-                                  np.mean(time_n))
-                break
-
-        return acq_val_incumbent, incumbent
+            for i, neighborhood_iterator in enumerate(neighborhood_iterators):
+                if active[i]:
+                    neighbors_for_i = []
+                    for j in range(obtain_n[i]):
+                        try:
+                            n = next(neighborhood_iterator)
+                            neighbors_generated[i] += 1
+                            neighbors_for_i.append(n)
+                        except StopIteration:
+                            obtain_n[i] = len(neighbors_for_i)
+                            new_neighborhood[i] = True
+                            break
+                    neighbors.extend(neighbors_for_i)
+
+            if len(neighbors) != 0:
+                start_time = time.time()
+                acq_val = self.acquisition_function(neighbors, **kwargs)
+                end_time = time.time()
+                times.append(end_time - start_time)
+                if np.ndim(acq_val.shape) == 0:
+                    acq_val = [acq_val]
+
+                # Comparing the acquisition function of the neighbors with the acquisition value of the incumbent
+                acq_index = 0
+                # Iterating the all i local searches
+                for i in range(num_incumbents):
+                    if not active[i]:
+                        continue
+                    # And for each local search we know how many neighbors we obtained
+                    for j in range(obtain_n[i]):
+                        # The next line is only true if there was an improvement and we basically need to iterate to
+                        # the i+1-th local search
+                        if improved[i]:
+                            acq_index += 1
+                        else:
+                            # Found a better configuration
+                            if acq_val[acq_index] > acq_val_incumbents[i]:
+                                incumbents[i] = neighbors[acq_index]
+                                acq_val_incumbents[i] = acq_val[acq_index]
+                                new_neighborhood[i] = True
+                                improved[i] = True
+                                local_search_steps[i] += 1
+                                neighbors_w_equal_acq[i] = []
+                                obtain_n[i] = 1
+                            # Found an equally well performing configuration, keeping it for plateau walking
+                            elif acq_val[acq_index] == acq_val_incumbents[i]:
```

Comment:
I don't get why you distinguish between plateau steps and improving steps here.

Reply:
It was the same before, I did not change the behavior here.